### PR TITLE
Added interpolation property to GPUParticles2D

### LINKED ISSUE #55357: Vulkan: GPUParticles2D interpolation not implemented yet (unlike in GPUParticles3D)
*Related to https://github.com/godotengine/godot/issues/51318.*

### Godot version

4.0.dev (5efe80f30)

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 470.74)

### Issue description

GPUParticles2D doesn't have an **Interpolate** property available. This helps make particles feel smoother without using too much GPU time, especially when attractors and collision are enabled. The effect is more noticeable on high refresh rate monitors, but now that GPUParticles simulation occurs at 30 Hz by default, interpolation is needed on 60 Hz monitors too to ensure smooth visuals.

### Steps to reproduce

- Add a GPUParticles2D node.
- Notice how there's no **Interpolate** property in the inspector.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by IgorKordiukiewicz**
Fixes #55357

### CODE REVIEW

File: doc/classes/GPUParticles2D.xml [hunk 1]
```diff
@@ -57,6 +57,8 @@
 		<member name="fract_delta" type="bool" setter="set_fractional_delta" getter="get_fractional_delta" default="true">
 			If [code]true[/code], results in fractional delta calculation which has a smoother particles display effect.
 		</member>
+		<member name="interpolate" type="bool" setter="set_interpolate" getter="get_interpolate" default="true">
+		</member>
```
Comment (KoBeWi) [inline doc/classes/GPUParticles2D.xml:61]:
```suggestion
		<member name="interpolate" type="bool" setter="set_interpolate" getter="get_interpolate" default="true">
			Enables particle interpolation, which makes the particle movement smoother when their [member fixed_fps] is lower than the screen refresh rate.
		</member>
```

**Review by KoBeWi — APPROVED** [review]
Looks good. I added a suggestion for the description to the property, but someone should verify if it's accurate.

File: doc/classes/GPUParticles2D.xml (hunk 1 above)
Comment (IgorKordiukiewicz) [reply doc/classes/GPUParticles2D.xml]: Updated

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**